library/tulip-core/src/Dikjstra.cpp

```diff
@@ -27,15 +27,20 @@ using namespace std;
 
 //============================================================
 Dikjstra::Dikjstra(const Graph *const graph, node src, const EdgeStaticProperty<double> &weights,
-                   NodeStaticProperty<double> &nd, std::function<Iterator<edge> *(node)> &getEdges)
-    : nodeDistance(nd) {
+                   NodeStaticProperty<double> &nd, std::stack<node> &qN, MutableContainer<int>& nP,
```

Comment:
The new members **queueNodes** and **numberOfPaths** are only needed when Dikjstra is instantiated in your new version of maxDistance. So they could be pointers instead of reference in order to not "compute" them when they are null. So the constructor arguments used to initialize them should be at the end and have nullptr default value.